core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -87,12 +88,13 @@ __global__ __launch_bounds__(
   const DriverType &driver = *(reinterpret_cast<const DriverType *>(
       kokkos_impl_hip_constant_memory_buffer));
 
-  driver->operator()();
+  driver();
```

Comment:
There's no real reason for this to be separate from the Cuda version, but if we're going to copy and paste things, let's at least add a comment that says something like "should be exactly the same as `cuda_parallel_launch_constant_memory()` and the analogous comment in `Kokkos_Cuda_KernelLaunch.hpp` so that anyone who changes either implementation knows to consider changing the other (or removing the comment, if appropriate).